### PR TITLE
feat: support cover title

### PR DESCRIPTION
**Pull request by LucasXu0**
### Feature Preview

closes https://github.com/AppFlowy-IO/AppFlowy/issues/6341

- [x] Remove the default name for view
- [x] Display the default name if the title is empty.
- [x] Add cover title
- [x] Support arrow keys to navigate between title and content
  - [x] arrow down and arrow right in title
  - [x] arrow up and arrow left in doc
- [x] Press enter key in title to jump to the first line of content (split the title text if the cursor is not at the end)
- [x] Replace the cover title selection color
- [x] Sync the name between cover title and view name
- [x] Press backspace key at the first line to go to the end of the title
- [ ] ~~Select all~~ (not supported yet)
- [x] Breaking change: The name of newly created pages is null or empty in this version if the user doesn't rename it. Users of the old version will see an empty name.
- [x] The backspace key doesn't work after pressing Enter in the title
- [ ] ~~Bring the paragraph to title when pressing backspace.~~ (blocking by date/reminder/linked page)
- [x] Only change the name for document, not including database

##### preview
https://github.com/user-attachments/assets/ebbe64bc-f778-470e-8796-7204c072414f

---

#### PR Checklist

- [x] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [x] I've listed at least one issue that this PR fixes in the description above.
- [x] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [ ] All existing tests are passing.

### CODE REVIEW

File: frontend/appflowy_flutter/integration_test/desktop/document/document_title_test.dart [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import 'package:integration_test/integration_test.dart';
+
+import '../../shared/util.dart';
+
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  group('document title', () {
+    testWidgets(
+        '''create a new document, the title will be displayed at the first line''',
```
Comment (Xazin) [inline frontend/appflowy_flutter/integration_test/desktop/document/document_title_test.dart:12]:
```suggestion
        'create a new document, the title will be displayed at the first line',
```

File: frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/base/cover_title_command.dart [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+final CommandShortcutEvent backspaceToTitle = CommandShortcutEvent(
+  key: 'backspace to title',
+  command: 'backspace',
+  getDescription: () => 'backspace to title',
+  handler: (editorState) => _backspaceToTitle(
+    editorState: editorState,
+  ),
+);
+
+KeyEventResult _backspaceToTitle({
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/base/cover_title_command.dart:20]: Shouldn't it also bring the paragraph up to the title?

File: frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/header/cover_title.dart [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+import 'package:appflowy/workspace/application/appearance_defaults.dart';
+import 'package:appflowy/workspace/application/view/view_bloc.dart';
+import 'package:appflowy_backend/protobuf/flowy-folder/view.pb.dart';
+import 'package:appflowy_editor/appflowy_editor.dart';
+import 'package:easy_localization/easy_localization.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+FocusNode? coverTitleFocusNode;
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/header/cover_title.dart:12]: I'm not a huge fan of this dangling global variable. Can we use an inherited widget or a Provider?

Comment (LucasXu0) [reply frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/header/cover_title.dart:12]: Let me see if there is a way to share this value. Using a provider or inherited widget doesn't work here unless we register this value at the top of the widget tree.

File: frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/base/cover_title_command.dart (hunk 2 above)
Comment (LucasXu0) [reply frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/base/cover_title_command.dart:21]: Yes. But I'm struggling with the reminder/date/linked page issue. These texts were formatted with $, it makes no sense to convert them back to $ when bringing the paragraph to title. :(

**Pull request merged**